### PR TITLE
Use BrainGlobe composite actions to run tests

### PR DESCRIPTION
**Pull request by dstansby**
This reduces duplication in the test configuration.

### CODE REVIEW

File: .github/workflows/linting_checks.yml [hunk 1]
```diff
@@ -1,41 +0,0 @@
...
-      - name: Install pre-commit hooks
-        run: pre-commit install
-
-      # This will run on all files in the repo not just those that have been
-      # committed. Once formatting has been applied once globally, from then on
-      # the files being changed by pre-commit should be just those that are
-      # being committed - provided that people are using the pre-commit hook to
-      # format their code.
-      - name: run pre-commit
-        run: pre-commit run --all-files --color always
```
Comment (paddyroddy) [inline .github/workflows/linting_checks.yml:41]: This was never meant to be here, I think someone messed up when doing a merge.

File: .github/workflows/test_and_deploy.yml [hunk 2]
```diff
@@ -64,34 +28,13 @@ jobs:
         python-version: [3.7, 3.8]
 
     steps:
-      - uses: actions/checkout@v2
-
-      - name: Set up Python ${{ matrix.python-version }}
-        uses: actions/setup-python@v2
+      # Setup pyqt libraries
+      - uses: tlambert03/setup-qt-libs@v1
```
Comment (paddyroddy) [inline .github/workflows/test_and_deploy.yml:32]: Why doesn't this get moved into the composite `test` action? Or is it only used in this repo?

**Review by paddyroddy — APPROVED** [review]
👍

File: .github/workflows/test_and_deploy.yml (hunk 2 above)
Comment (dstansby) [reply .github/workflows/test_and_deploy.yml:32]: The test action is meant to be generic enough to be used without setting up QT libraries, e.g. `cellfinder-core` doesn't need to use setup QT steps.

**Pull request merged**